diff --git a/drivers/android/vendor_hooks.c b/drivers/android/vendor_hooks.c
index 1cd4d5f94..d06087a55 100644
--- a/drivers/android/vendor_hooks.c
+++ b/drivers/android/vendor_hooks.c
@@ -209,10 +209,8 @@ EXPORT_TRACEPOINT_SYMBOL_GPL(android_rvh_irqs_disable);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_rvh_irqs_enable);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_is_fpsimd_save);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_slab_folio_alloced);
-EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_madvise_swapin_walk_pmd_entry);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_process_madvise_begin);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_process_madvise_iter);
-EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_process_madvise_end);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_kmalloc_large_alloced);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_tune_swappiness);
 EXPORT_TRACEPOINT_SYMBOL_GPL(android_vh_sched_show_task);
diff --git a/include/trace/hooks/mm.h b/include/trace/hooks/mm.h
index f76fb3498..979a058bd 100644
--- a/include/trace/hooks/mm.h
+++ b/include/trace/hooks/mm.h
@@ -36,18 +36,12 @@ DECLARE_HOOK(android_vh_si_meminfo_adjust,
 DECLARE_HOOK(android_vh_slab_folio_alloced,
 	TP_PROTO(unsigned int order, gfp_t flags),
 	TP_ARGS(order, flags));
-DECLARE_HOOK(android_vh_madvise_swapin_walk_pmd_entry,
-	TP_PROTO(swp_entry_t entry),
-	TP_ARGS(entry));
 DECLARE_HOOK(android_vh_process_madvise_begin,
 	TP_PROTO(struct task_struct *task, int behavior),
 	TP_ARGS(task, behavior));
 DECLARE_HOOK(android_vh_process_madvise_iter,
 	TP_PROTO(struct task_struct *task, int behavior, ssize_t *ret),
 	TP_ARGS(task, behavior, ret));
-DECLARE_HOOK(android_vh_process_madvise_end,
-	TP_PROTO(struct task_struct *task, int behavior, ssize_t *ret),
-	TP_ARGS(task, behavior, ret));
 DECLARE_HOOK(android_vh_kmalloc_large_alloced,
 	TP_PROTO(struct page *page, unsigned int order, gfp_t flags),
 	TP_ARGS(page, order, flags));
diff --git a/mm/madvise.c b/mm/madvise.c
index 4eb44e3ed..324dc91cf 100644
--- a/mm/madvise.c
+++ b/mm/madvise.c
@@ -219,7 +219,6 @@ static int swapin_walk_pmd_entry(pmd_t *pmd, unsigned long start,
 
 		pte_unmap_unlock(ptep, ptl);
 		ptep = NULL;
-		trace_android_vh_madvise_swapin_walk_pmd_entry(entry);
 
 		page = read_swap_cache_async(entry, GFP_HIGHUSER_MOVABLE,
 					     vma, addr, &splug);
@@ -1552,14 +1551,16 @@ SYSCALL_DEFINE5(process_madvise, int, pidfd, const struct iovec __user *, vec,
 	trace_android_vh_process_madvise_begin(task, behavior);
 
 	while (iov_iter_count(&iter)) {
+		trace_android_vh_process_madvise_iter(task, behavior, &ret);
+		if (ret < 0)
+			break;
 		ret = do_madvise(mm, (unsigned long)iter_iov_addr(&iter),
 					iter_iov_len(&iter), behavior);
-		trace_android_vh_process_madvise_iter(task, behavior, &ret);
 		if (ret < 0)
 			break;
 		iov_iter_advance(&iter, iter_iov_len(&iter));
 	}
-	trace_android_vh_process_madvise_end(task, behavior, &ret);
+
 	ret = (total_len - iov_iter_count(&iter)) ? : ret;
 
 release_mm:
