diff --git a/android/abi_gki_aarch64.stg b/android/abi_gki_aarch64.stg
index 1494e87b054b..bec129964f10 100644
--- a/android/abi_gki_aarch64.stg
+++ b/android/abi_gki_aarch64.stg
@@ -203,6 +203,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0x28e34a3d
 }
+pointer_reference {
+  id: 0x00abf683
+  kind: POINTER
+  pointee_type_id: 0x28ef3c91
+}
 pointer_reference {
   id: 0x00b7947f
   kind: POINTER
@@ -2993,6 +2998,11 @@ pointer_reference {
   kind: POINTER
   pointee_type_id: 0x00a70faf
 }
+pointer_reference {
+  id: 0x0abac407
+  kind: POINTER
+  pointee_type_id: 0x00abf683
+}
 pointer_reference {
   id: 0x0ac23bc5
   kind: POINTER
@@ -33143,6 +33153,11 @@ typedef {
   name: "pfn_t"
   referred_type_id: 0x0b8d77a1
 }
+typedef {
+  id: 0x28ef3c91
+  name: "pg_data_t"
+  referred_type_id: 0x264eeece
+}
 typedef {
   id: 0xaf12ea2c
   name: "pgd_t"
@@ -307824,6 +307839,15 @@ function {
   parameter_id: 0x1a41d386
   parameter_id: 0x3fa905dc
 }
+function {
+  id: 0x16f8db1d
+  return_type_id: 0x48b5725f
+  parameter_id: 0x18bd6530
+  parameter_id: 0x13580d6c
+  parameter_id: 0x0abac407
+  parameter_id: 0x064d6086
+  parameter_id: 0x11cfee5a
+}
 function {
   id: 0x16fd178e
   return_type_id: 0x3015c47d
@@ -308451,6 +308475,11 @@ function {
   return_type_id: 0x48b5725f
   parameter_id: 0x1d5934ce
 }
+function {
+  id: 0x17ceba19
+  return_type_id: 0x48b5725f
+  parameter_id: 0x1d5bae2a
+}
 function {
   id: 0x17d0688a
   return_type_id: 0x48b5725f
@@ -328149,6 +328178,13 @@ function {
   parameter_id: 0x09a83f1c
   parameter_id: 0x68afbc33
 }
+function {
+  id: 0x958c659d
+  return_type_id: 0x6720d32f
+  parameter_id: 0x2170d06d
+  parameter_id: 0x1b36c7a2
+  parameter_id: 0xf1a6dfed
+}
 function {
   id: 0x958c8293
   return_type_id: 0x6720d32f
@@ -352751,6 +352787,24 @@ elf_symbol {
   type_id: 0x901c3c62
   full_name: "__media_entity_setup_link"
 }
+elf_symbol {
+  id: 0x9191cfa1
+  name: "__mem_cgroup_charge"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x8cf6a2f9
+  type_id: 0x958c659d
+  full_name: "__mem_cgroup_charge"
+}
+elf_symbol {
+  id: 0x9090edf7
+  name: "__mem_cgroup_uncharge"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xe81b0cf4
+  type_id: 0x18c46588
+  full_name: "__mem_cgroup_uncharge"
+}
 elf_symbol {
   id: 0x47a4997d
   name: "__memcat_p"
@@ -352857,6 +352911,15 @@ elf_symbol {
   type_id: 0x1e768676
   full_name: "__mod_lruvec_state"
 }
+elf_symbol {
+  id: 0xeaad658b
+  name: "__mod_memcg_lruvec_state"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x79f363e6
+  type_id: 0x1e768676
+  full_name: "__mod_memcg_lruvec_state"
+}
 elf_symbol {
   id: 0x9de4c040
   name: "__mod_zone_page_state"
@@ -369489,6 +369552,15 @@ elf_symbol {
   type_id: 0x9af01625
   full_name: "btbcm_setup_patchram"
 }
+elf_symbol {
+  id: 0x687bbce2
+  name: "bucket_order"
+  is_defined: true
+  symbol_type: OBJECT
+  crc: 0xd5ecc56f
+  type_id: 0x4585663f
+  full_name: "bucket_order"
+}
 elf_symbol {
   id: 0x6bb6d4ca
   name: "buffer_migrate_folio"
@@ -395483,6 +395555,15 @@ elf_symbol {
   type_id: 0x19ccd2d6
   full_name: "media_request_put"
 }
+elf_symbol {
+  id: 0xab391a40
+  name: "mem_cgroup_flush_stats_ratelimited"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0x91af08a7
+  type_id: 0x17ceba19
+  full_name: "mem_cgroup_flush_stats_ratelimited"
+}
 elf_symbol {
   id: 0x140f40dd
   name: "mem_cgroup_from_id"
@@ -419042,6 +419123,15 @@ elf_symbol {
   type_id: 0x156e1138
   full_name: "unmap_mapping_range"
 }
+elf_symbol {
+  id: 0xa86562be
+  name: "unpack_shadow"
+  is_defined: true
+  symbol_type: FUNCTION
+  crc: 0xa40438df
+  type_id: 0x16f8db1d
+  full_name: "unpack_shadow"
+}
 elf_symbol {
   id: 0x60ffe144
   name: "unpin_user_page"
@@ -426126,6 +426216,8 @@ interface {
   symbol_id: 0x1dfbf9ac
   symbol_id: 0x91810af1
   symbol_id: 0xc50a781f
+  symbol_id: 0x9191cfa1
+  symbol_id: 0x9090edf7
   symbol_id: 0x47a4997d
   symbol_id: 0x20b2f551
   symbol_id: 0x92f8c9de
@@ -426138,6 +426230,7 @@ interface {
   symbol_id: 0x0ba13d85
   symbol_id: 0x603dcc7d
   symbol_id: 0x68d1b187
+  symbol_id: 0xeaad658b
   symbol_id: 0x9de4c040
   symbol_id: 0x89d01efd
   symbol_id: 0xf680e61c
@@ -427986,6 +428079,7 @@ interface {
   symbol_id: 0xfc8c52c1
   symbol_id: 0xb47c5ffe
   symbol_id: 0xdded4dcd
+  symbol_id: 0x687bbce2
   symbol_id: 0x6bb6d4ca
   symbol_id: 0x8ea00f54
   symbol_id: 0x0f9d7b5e
@@ -430866,6 +430960,7 @@ interface {
   symbol_id: 0xd075c3be
   symbol_id: 0xa848deda
   symbol_id: 0x8eadb5fd
+  symbol_id: 0xab391a40
   symbol_id: 0x140f40dd
   symbol_id: 0x6da682ae
   symbol_id: 0x7f797603
@@ -433482,6 +433577,7 @@ interface {
   symbol_id: 0xcad66794
   symbol_id: 0x3cace439
   symbol_id: 0xe41b8f90
+  symbol_id: 0xa86562be
   symbol_id: 0x60ffe144
   symbol_id: 0xffd70606
   symbol_id: 0xe19f4159
